include/exadg/operators/operator_base.cpp

```diff
@@ -305,28 +313,42 @@ template<int dim, typename Number, int n_components>
 void
 OperatorBase<dim, Number, n_components>::apply_add(VectorType & dst, VectorType const & src) const
 {
-  if(is_dg)
+  if(evaluate_face_integrals())
   {
-    if(evaluate_face_integrals())
+    if(is_dg)
     {
       matrix_free->loop(
         &This::cell_loop, &This::face_loop, &This::boundary_face_loop_hom_operator, this, dst, src);
     }
     else
     {
-      matrix_free->cell_loop(&This::cell_loop, this, dst, src);
+      // Compute matrix-vector product. Constrained degrees of freedom in the `src`-vector will not
+      // be used. The function `read_dof_values()` (or `gather_evaluate()`) uses the homogeneous
+      // boundary data passed to `MatrixFree` via `AffineConstraints` with the standard `dof_index`.
+      matrix_free->loop(&This::cell_loop,
+                        &This::face_loop_empty,
+                        &This::boundary_face_loop_hom_operator,
+                        this,
+                        dst,
+                        src);
     }
   }
   else
   {
-    // See function apply() for additional comments.
-    // Note that MatrixFree will not touch constrained degrees of freedom in the dst-vector.
+    // See function `apply()` for additional comments. Note that `dealii::MatrixFree` will not touch
+    // constrained degrees of freedom in the `dst` vector.
     matrix_free->cell_loop(&This::cell_loop, this, dst, src);
+  }
 
+  if(not is_dg)
+  {
+    // Constrained degrees of freedom are not removed from the system of equations. Instead, we set
+    // the diagonal entries of the matrix to 1 for these constrained degrees of freedom. This means
+    // that we need to copy the constrained values to the `dst` vector.
     for(unsigned int const constrained_index :
         matrix_free->get_constrained_dofs(this->data.dof_index))
     {
-      dst.local_element(constrained_index) += src.local_element(constrained_index);
+      dst.local_element(constrained_index) = src.local_element(constrained_index);
```

Comment:
I don't understand this change. The function is `apply_add`, so shouldn't we keep the code line as it was before?